apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -32,6 +37,12 @@ pub struct Oracle {
     /// A channel to signal that the ledger should shut down
     /// because the Oracle has stopped
     abort: Option<Sender<()>>,
+    /// How long the oracle should wait between checking blocks
+    backoff: Duration,
+    /// If provided, the oracle will attempt to send block heights that it has
+    /// processed to this channel, but won't pause if this channel is full
+    /// or disconnected.
+    blocks_processed: Option<BoundedSender<Uint256>>,
```

Comment:
We don't initialize this in production right now, only for tests so we can check what blocks the oracle has actually processed. Alternatively we could have something like:

```rust
latest_block_processed: Arc<Mutex<Uint256>>
```

i.e. something which lets us look at what is the latest block height that the oracle has processed, from any thread

Reply:
https://docs.rs/tokio/latest/tokio/sync/watch/index.html
use a single producer, multiple consumer buffer?

Reply:
Is there an obvious use case where a ledger would need this?

Reply:
If it's just for testing, we can use a oneshot from the mock web3 client like we do for checking the events have been processed.

Reply:
No obvious use case for the ledger - I think we can dispatch the oracle to do it's thing without the ledger having to know which individual block(s) it has processed. Publishing Ethereum blocks processed metrics could be useful but not a priority right now. We need some way to observe which block(s) have been processed for the tests, the `tokio::sync::watch` looks like it would be suitable here although I'm finding it fiddlier (currently trying it out), I'm not sure how to do it with a `oneshot`?

Reply:
I reverted to the original use of a `BoundedSender` on which we send all block heights that we process, so that we can check every one for `test_all_blocks_checked`